pkg/ebpf/c/tracer-ebpf.c

```diff
@@ -634,19 +659,61 @@ int kretprobe__tcp_close(struct pt_regs* ctx) {
     return 0;
 }
 
-SEC("kprobe/udp_sendmsg")
-int kprobe__udp_sendmsg(struct pt_regs* ctx) {
+SEC("kprobe/ip6_make_skb")
+int kprobe__ip6_make_skb(struct pt_regs* ctx) {
+     struct sock* sk = (struct sock*)PT_REGS_PARM1(ctx);
+     size_t size = (size_t)PT_REGS_PARM4(ctx);
+     u64 pid_tgid = bpf_get_current_pid_tgid();
+
+     size = size - sizeof(struct udphdr);
+
+     conn_tuple_t t = {};
+     if (!read_conn_tuple(&t, sk, pid_tgid, CONN_TYPE_UDP)) {
+
+         increment_telemetry_count(udp_send_missed);
+         return 0;
+     }
+
+     log_debug("kprobe/ip6_make_skb: pid_tgid: %d, size: %d\n", pid_tgid, size);
+     handle_message(&t, size, 0);
+     increment_telemetry_count(udp_send_processed);
+
+     return 0;
+}
+
+SEC("kprobe/ip_make_skb")
```

Comment:
curious, could we use this kprobe for both TCP and UDP? it seems that this would triggered for both, right?

Reply:
As far as I know, it works only for UDP.
From. https://elixir.bootlin.com/linux/latest/ident/ip_make_skb:

<img width="347" alt="Screen Shot 2020-09-04 at 9 56 03 AM" src="https://user-images.githubusercontent.com/6791023/92253229-dd160600-ee94-11ea-98a6-f92cbc33dbeb.png">

Reply:
Gotcha. Another thing worth checking is if the arguments of this function haven't changed since kernel 4.4.0

Reply:
My understanding so far is that unlike UDP, TCP is not limited to one method for creating SKB. It calls different functions depending on the context (e.g., SKBs for retransmission are treated different than a regular transmission).

Reply:
> could we use this kprobe for both TCP and UDP

It might be worth adding a comment somewhere that this is only used in the UDP code path. I remember the name of the function causing me to ask the same question.